schema_salad/python_codegen.py

```diff
@@ -79,30 +79,49 @@ def begin_class(self,  # pylint: disable=too-many-arguments
             self.out.write("    pass\n\n")
             return
 
-        self.out.write(
-            """    def __init__(self, _doc, baseuri, loadingOptions, docRoot=None):
-        # type: (Any, Text, LoadingOptions, Optional[Text]) -> None
-        doc = copy.copy(_doc)
-        if hasattr(_doc, 'lc'):
-            doc.lc.data = _doc.lc.data
-            doc.lc.filename = _doc.lc.filename
-        errors = []
+        safe_inits = ["self"]  # type: List[Text]
+        safe_inits.extend([self.safe_name(f) for f in field_names if f != "class"])
+        self.out.write("    def __init__(" +", ".join(safe_inits) + """, extension_fields=None, loadingOptions=None):
```

Comment:
Needs to emit a type signature to make MyPy happy

Reply:
Yeah, gonna have to be `All` for each, for now (unless you have time to code up a proper type signature generator)